modules/imgcodecs/src/loadsave.cpp

```diff
@@ -631,15 +631,19 @@ bool imreadmulti(const String& filename, std::vector<Mat>& mats, int start, int
 {
     CV_TRACE_FUNCTION();
 
+    int counter = 0;
     for (int i = start; i < start+count; i++)
     {
         Mat page;
         imread_(filename, flags, page, i);
 
         if (!page.empty())
+        {
+            counter++;
             mats.push_back(page);
+        }
     }
-    return !mats.empty();
+    return counter == count;
```

Comment:
If we asked for more images than multipage page count and we successfully read existing images should this be false ?